### PR TITLE
Move release process to be based off of tagging instead of pushes to main

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,257 @@
+name: release
+
+on:
+  push:
```
Comment (ekoby) [inline .github/workflows/release.yml:4]: not release trigger?

Comment (plorenz) [reply .github/workflows/release.yml:4]: Tag creates the release, marked as pre-release. Could do it the other way, but it's even more manual

File: RELEASING.md [hunk 2]
```diff
@@ -59,12 +60,24 @@ Perform these steps in PR branches based on release-next (trunk).
...
+1. Push a tag like v*, typically on default branch HEAD to trigger the pre-release workflow named `release.yml`.
+
+## Stable and Latest Release
+
+Pre-releases are releases, but they're not promoted as "latest" in GitHub or automatically shipped downstream. Marking a
+release as not a prerelease makes it a stable release. There can be one stable release that's also marked "latest"
+(`isLatest: true`).
+
+1. After an arbitrary burn-in period, unmark "prerelease" in GitHub Releases (`isPrerelease: false`) to stop
+   discouraging users from consuming the now-stable release. This will automatically promote and advertise the downstreams.
```
Comment (ekoby) [inline RELEASING.md:78]: is this supposed to be `non-stable`?

`stop discouraging` reads like double negative, I had to think what it means

**Review by ekoby — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (ekoby) [reply .github/workflows/release.yml:4]: don't we want release to be a deliberate manual process? 
either way is fine

Comment (plorenz) [reply .github/workflows/release.yml:4]: It is a deliberate manual process, it's just one done via tag, push tag, rather than through the GH UI.

**Review by qrkourier — APPROVED** [review]

**Pull request merged**